File: wenger-common/src/main/java/io/github/rcarlosdasilva/wenger/common/exception/WengerException.java
package io.github.rcarlosdasilva.wenger.common.exception;

public class WengerException extends Exception {

  public WengerException() {
    super();
  }

  public WengerException(String message) {
    super(message);
  }

  public WengerException(String message, Throwable cause) {
    super(message, cause);
  }

  public WengerException(Throwable cause) {
    super(cause);
  }

  protected WengerException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
    super(message, cause, enableSuppression, writableStackTrace);
  }

}

File: wenger-common/src/main/java/io/github/rcarlosdasilva/wenger/common/exception/WengerException.kt
package io.github.rcarlosdasilva.wenger.common.exception

open class WengerException : Exception {
  constructor() : super()
  constructor(message: String?) : super(message)
  constructor(message: String?, cause: Throwable?) : super(message, cause)
  constructor(cause: Throwable?) : super(cause)
  constructor(message: String?, cause: Throwable?, enableSuppression: Boolean, writableStackTrace: Boolean) : super(message, cause, enableSuppression, writableStackTrace)
}

open class WengerRuntimeException : RuntimeException {
  constructor() : super()
  constructor(message: String?) : super(message)
  constructor(message: String?, cause: Throwable?) : super(message, cause)
  constructor(cause: Throwable?) : super(cause)
  constructor(message: String?, cause: Throwable?, enableSuppression: Boolean, writableStackTrace: Boolean) : super(message, cause, enableSuppression, writableStackTrace)
}